x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/latest/LatestDoc.java

```diff
@@ -0,0 +1,280 @@
+        if (aggregations == null) {
+            return null;
+        }
+
+        CompositeAggregation compositeAgg = aggregations.get(COMPOSITE_AGGREGATION_NAME);
+        if (compositeAgg == null || compositeAgg.getBuckets().isEmpty()) {
+            return null;
+        }
+
+        Stream<IndexRequest> indexRequestStream =
+            extractResults(compositeAgg, stats)
+                .map(document -> DocumentConversionUtils.convertDocumentToIndexRequest(document, destinationIndex, destinationPipeline));
+        return Tuple.tuple(indexRequestStream, compositeAgg.afterKey());
+    }
+
+    @Override
+    public void validateQuery(Client client, SourceConfig sourceConfig, ActionListener<Boolean> listener) {
+        SearchRequest searchRequest = buildSearchRequest(sourceConfig, null, TEST_QUERY_PAGE_SIZE);
+
+        client.execute(SearchAction.INSTANCE, searchRequest, ActionListener.wrap(response -> {
+            if (response == null) {
+                listener.onFailure(
+                    new ElasticsearchStatusException("Unexpected null response from test query", RestStatus.SERVICE_UNAVAILABLE)
+                );
+                return;
+            }
+            if (response.status() != RestStatus.OK) {
+                listener.onFailure(
+                    new ElasticsearchStatusException(
+                        "Unexpected status from response of test query: " + response.status(),
+                        response.status()
+                    )
+                );
+                return;
+            }
+            listener.onResponse(true);
+        }, e -> {
+            Throwable unwrapped = ExceptionsHelper.unwrapCause(e);
+            RestStatus status = unwrapped instanceof ElasticsearchException
+                ? ((ElasticsearchException) unwrapped).status()
+                : RestStatus.SERVICE_UNAVAILABLE;
+            listener.onFailure(new ElasticsearchStatusException("Failed to test query", status, unwrapped));
+        }));
+    }
+
+    @Override
+    public void validateConfig(ActionListener<Boolean> listener) {
+        listener.onResponse(true);
+    }
+
+    @Override
+    public void deduceMappings(Client client, SourceConfig sourceConfig, ActionListener<Map<String, String>> listener) {
+        FieldCapabilitiesRequest fieldCapabilitiesRequest =
+            new FieldCapabilitiesRequest().indices(sourceConfig.getIndex())
+                .fields("*")
+                .indicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN);
+        client.execute(
+            FieldCapabilitiesAction.INSTANCE,
+            fieldCapabilitiesRequest,
+            ActionListener.wrap(
+                response -> listener.onResponse(
+                    SchemaUtil.extractFieldMappings(response).entrySet().stream()
+                        .filter(not(e -> e.getKey().startsWith("_")))
+                        .collect(toMap(Map.Entry::getKey, Map.Entry::getValue))
```

Comment:
same concern as before around user defined fields that start with `_`

Reply:
Needs investigation, added TODO in the `DocumentConversionUtils.removeInternalFields` method comment.